### PR TITLE
Update ecl wells for all matching wells after actionx

### PR DESCRIPTION
**Pull request by joakim-hove**

### CODE REVIEW

**Comment by joakim-hove** [comment]
jenkins build this please

**Comment by joakim-hove** [comment]
jenkins build this please

File: opm/simulators/wells/BlackoilWellModel_impl.hpp [hunk 1]
```diff
@@ -2652,29 +2652,54 @@ namespace Opm {
...
-        }
 
-        this->well_state_.resetConnectionTransFactors(well_index, pd);
-        this->prod_index_calc_[well_index].reInit(well);
+            this->well_state_.resetConnectionTransFactors(wi, pd);
+            this->prod_index_calc_[wi].reInit(well);
+        }
+        const auto& comm = this->ebosSimulator_.vanguard().grid().comm();
+        this->well_state_.updateGlobalIsGrup(schedule, timeStepIdx, comm);
     }
```
Comment (totto82) [inline opm/simulators/wells/BlackoilWellModel_impl.hpp:2703]: I suggest to do a full `updateAndCommunicateGroupData` here not just update the global Grup.

File: opm/simulators/wells/BlackoilWellModel_impl.hpp [hunk 2]
```diff
@@ -2652,29 +2652,54 @@ namespace Opm {
...
-                pdIter->connection_transmissibility_factor = conn.CF();
-                ++pdIter;
+            this->well_state_.updateStatus(wi, well.getStatus());
+            if (well.getStatus() == Well::Status::OPEN) {
+                printf("Opening well: injector: %d  %s\n", well.isInjector(), well.name().c_str());
+                if (well.isInjector()) {
+                    auto controls = this->well_state_.currentInjectionControls();
+                    printf("Control: %d -> ", static_cast<int>(controls[wi]));
+                    if (controls[wi] == Well::InjectorCMode::CMODE_UNDEFINED)
+                        controls[wi] = well.injection_cmode();
```
Comment (totto82) [inline opm/simulators/wells/BlackoilWellModel_impl.hpp:2689]: Does this if trigger at all?

**Comment by joakim-hove** [comment]
jenkins build this please

**Comment by joakim-hove** [comment]
jenkins build this please

**Comment by joakim-hove** [comment]
jenkins build this please

**Comment by joakim-hove** [comment]
jenkins build this please

**Comment by joakim-hove** [comment]
jenkins build this please

File: opm/simulators/wells/BlackoilWellModel_impl.hpp (hunk 1 above)
Comment (totto82) [reply opm/simulators/wells/BlackoilWellModel_impl.hpp:2667]: Sure you need the `updateGlobalIsGrup`. I removed it get the same results. 
The `applyActions` is called in `endTimeStep` and `updateAndCommunicateGroupData` (where `updateGlobalIsGrup` is called) is called in `beginTimeStep`. Nothing well related is happening in between.

File: opm/simulators/wells/WellInterface_impl.hpp [hunk 3]
```diff
@@ -2230,7 +2230,7 @@ namespace Opm
         WellGroupHelpers::FractionCalculator fcalc(schedule, summaryState, well_state, current_step_, guide_rate_, tcalc.guideTargetMode(), pu, false, injectionPhase);
 
         auto localFraction = [&](const std::string& child) {
-            return fcalc.localFraction(child, "");
+            return fcalc.localFraction(child, child);
         };
```
Comment (totto82) [inline opm/simulators/wells/WellInterface_impl.hpp:2234]: This change shouldnt be needed. Child is already on grup control and you don't need to pass its name here. ACTIONX_M1 runs perfectly without it.

File: opm/simulators/wells/BlackoilWellModel_impl.hpp (hunk 1 above)
Comment (joakim-hove) [reply opm/simulators/wells/BlackoilWellModel_impl.hpp:2667]: No I am not sure; I was trying so many things. Your argument is good - I will remove it.

File: opm/simulators/wells/WellInterface_impl.hpp (hunk 3 above)
Comment (joakim-hove) [reply opm/simulators/wells/WellInterface_impl.hpp:2234]: Ehhh - no that I do not agree. It runs through also in master, but the problem is that injection in well `WI1` starts to late. The ACTIONX triggers at 2020-07-20 and injection should start immediately after that, if I remove `(child,child)` change injection will not start before the next report step - at 2020-08-01.

Comment (totto82) [reply opm/simulators/wells/WellInterface_impl.hpp:2234]: Removing this did not change anything for me. Results, number of iterations etc. are the same.

**Comment by joakim-hove** [comment]
jenkins build this please

File: opm/simulators/wells/WellInterface_impl.hpp (hunk 3 above)
Comment (joakim-hove) [reply opm/simulators/wells/WellInterface_impl.hpp:2234]: You should run flow with `--solver-max-time-step-in-days=1` to ensure that the ACTIONX triggers well before the report step ends.

**Comment by joakim-hove** [comment]
jenkins build this please

**Comment by joakim-hove** [comment]
jenkins build this please

**Comment by joakim-hove** [comment]
jenkins build this please

**Comment by joakim-hove** [comment]
jenkins build this please

**Pull request merged**